src/main/java/org/jabref/logic/exporter/SaveSession.java

```diff
@@ -49,4 +55,14 @@ public void commit(String path) throws SaveException {
     public void addFieldChanges(List<FieldChange> newUndoableFieldChanges) {
         this.undoableFieldChanges.addAll(newUndoableFieldChanges);
     }
+
+    public void finalize(Path file) throws SaveException, IOException {
```

Comment:
Just a thought: Would it make sense to have this class implement `AutoCloseable`, so that it could be used in a try-with-resources? This method seems to be built for that use case. Otherwise, it's not guaranteed that the writer will be closed.

Reply:
This is a good idea but there are quite a few places where the save session is cancel (i.e. `commit` or `finalized` is not called). Moreover, a quick look at the code showed that it is relatively hard to shift the `file` argument to the constructor of the save session (the creation happens in different classes then the actual commit).